continuous_integration/gpuci/environment-3.10.yaml

```
@@ -15,7 +15,7 @@ dependencies:
 - intake>=0.6.0
 - jsonschema
 - lightgbm
-- maturin>=1.1,<1.2
```

Just out of interest, what are the specific features we need from the newer version of maturin? Also might make sense to pin this a little more tightly if the specific version we need is 1.3.1:


```suggestion
- maturin>=1.3.1,<1.4
```